packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts

```diff
@@ -358,7 +358,47 @@ export const a11y: AuditRuleWithSelector[] = [
 		match(element: HTMLElement) {
 			// innerText is used to ignore hidden text
 			const innerText = element.innerText.trim();
-			if (innerText === '') return true;
+			// Check for aria-label
+			const ariaLabel = element.getAttribute('aria-label')?.trim();
+			// Check for valid aria-labelledby
+			let ariaLabelledbyValid = false;
+			const ariaLabelledby = element.getAttribute('aria-labelledby')?.trim();
+			if (ariaLabelledby) {
+				const ids = ariaLabelledby.split(' ');
+				// Check if at least one of the referenced elements exists and has non-empty text
+				ariaLabelledbyValid = ids.some((id) => {
+					const referencedElement = document.getElementById(id);
+					return referencedElement && referencedElement.innerText.trim() !== '';
+				});
+			}
+			// Check for <img> with valid alt attribute
+			let imgWithAlt = false;
+			const imgElements = element.querySelectorAll('img');
+			for (const img of imgElements) {
+				const altAttribute = img.getAttribute('alt');
+				if (altAttribute && altAttribute.trim() !== '') {
+					imgWithAlt = true;
+					break;
+				}
+			}
+			// Check for <svg> with valid title
+			let svgWithTitle = false;
+			const svgElements = element.querySelectorAll('svg');
+			for (const svg of svgElements) {
+				const titleText = svg.querySelector('title');
+				if (titleText && titleText.textContent && titleText.textContent.trim() !== '') {
+					svgWithTitle = true;
+					break;
+				}
+			}
+			// Return true if innerText, ariaLabel, ariaLabelledby, imgWithAlt and svgWithTitle are all empty or invalid
+			return (
+				innerText === '' &&
+				(!ariaLabel || ariaLabel === '') &&
+				!ariaLabelledbyValid &&
+				!imgWithAlt &&
+				!svgWithTitle
+			);
```

Comment:
Thanks for including svg title 👏 

I'd suggest returning early in each check. Unnecessary processing for most elements. Personally I find this larger return statement with negates harder to read.

Reply:
That's a good point thank you